api/v1alpha1/authorization_types.go

```diff
@@ -72,6 +76,20 @@ type Principal struct {
 	// JWT authentication in the same `SecurityPolicy`.
 	// +optional
 	JWT *JWTPrincipal `json:"jwt,omitempty"`
+
+	// Headers authorize the request based on the headers in the request.
+	// If multiple headers are specified, all headers must match for the rule to match.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +notImplementedHide
+	Headers []HeaderMatch `json:"headers,omitempty"`
+
+	// Methods authorize the request based on the HTTP methods.
+	// If multiple methods are specified, one of the methods must match for the rule to match.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +notImplementedHide
+	Methods []gwapiv1.HTTPMethod `json:"methods,omitempty"`
```

Comment:
should methods live in the top level outside `principal` because its falls under `actions` / `operation` performed by the principal

Reply:
Yeah, moving `methods` outside of `principal` looks better.

Reply:
referencing istio here https://istio.io/latest/docs/reference/config/security/authorization-policy/#Operation
should we have a top level `operation` here ?
so we can have `operation.methods` and leave room for `operation.paths` in the future ?